app/components/polls/statements/Cards.tsx

```diff
@@ -44,12 +65,9 @@ const Cards = ({
   // render, because that would cause the cards to jump around. So we keep track of the
   // randomized order in state.
   const [statementSorting, setStatementSorting] = useState<number[]>([]);
+
   useEffect(() => {
-    setStatementSorting(
-      statements
-        .map((statement) => statement.id)
-        .sort(() => 0.5 - Math.random()),
-    );
+    setStatementSorting(getStatementSorting(statements));
     // eslint-disable-next-line react-hooks/exhaustive-deps
   }, []);
```

Comment:
This is not your code but I'd still like to fix it while we're working on this. There's no need to populate `statementSorting` in a useEffect when we have everything we need on the first render. All of this can just be:

```tsx
 const [statementSorting] = useState<number[]>(
    getStatementSorting(statements),
  );
```

Reply:
True. 

I did this first. Then moved the sorting to server side, encountered some issues and reverted back here.

I'll update it.